File: fixtures/scale/elixir/large/lib/core/benchmark.ex
defmodule Core.Benchmark do
  alias Support.Result
  alias Support.Timer

  defstruct name: "", elapsed: 0, green: 0

  def run(name, results) do
    started = Timer.start()
    green = Enum.count(results, &Result.ok?/1)
    %__MODULE__{name: name, elapsed: Timer.measure(started), green: green}
  end
end

File: fixtures/scale/haskell/large/Core/Benchmark.hs
module Core.Benchmark (Bench(..), run) where

import Support.Result (TestResult, ok)
import Support.Timer (Span, measure)

data Bench = Bench { benchName :: String, elapsed :: Span, greenCount :: Int }

run :: String -> Int -> Int -> [TestResult] -> Bench
run n start end rs = Bench n (measure start end) (length (filter ok rs))
